### PR TITLE
Add gen_hdf_file.cpp to support create random dataset

### PR DESCRIPTION
**Pull request by cydrain**

### CODE REVIEW

**Review by zhengbuqian — COMMENTED** [review]
thanks!

File: benchmark/utils.h [hunk 1]
```diff
@@ -82,3 +82,15 @@ GenRandomBitset(size_t n, size_t t) {
     }
     return data;
 }
+
+inline float*
+GenRandomVector(size_t rows, size_t dim) {
+    std::mt19937 rng(42);
+    // std::uniform_int_distribution<> distrib(0.0, 100.0);
```
[the remark this answers is not in the document]
Comment (cydrain) [reply benchmark/utils.h:89]: remove this API

File: benchmark/hdf5/gen_hdf5_file.cpp [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+#include "knowhere/comp/brute_force.h"
+#include "knowhere/comp/index_param.h"
+#include "knowhere/comp/knowhere_config.h"
+#include "knowhere/dataset.h"
+
+knowhere::DataSetPtr
+GenDataSet(int rows, int dim) {
+    std::mt19937 rng(42);
+    std::uniform_real_distribution<> distrib(-1.0, 1.0);
+    float* ts = new float[rows * dim];
```
[the remark this answers is not in the document]
Comment (cydrain) [reply benchmark/hdf5/gen_hdf5_file.cpp:27]: GenRandomVector removed

**Pull request merged**